subcommittee/tooling/tool-list/available-tools.yaml

```
@@ -0,0 +1,210 @@
+# This file lists available safety-critical Rust tools and their qualification state.
+
+metadata:
+  title: "Rust Tooling for use in Safety Critical Projects"
+  version: "1.0"
+  date: "2025-09-29"
+  tracked-standards:
+    - name: "ISO-26262"
+      levels: ["ASIL-A", "ASIL-B", "ASIL-C", "ASIL-D"]
+      description: "Road vehicles - Functional safety"
+    - name: "DO-178C"
+      levels: ["DAL-E", "DAL-D", "DAL-C", "DAL-B", "DAL-A"]
+      description: "Software Considerations in Airborne Systems and Equipment Certification"
+    - name: "IEC 62304"
+      levels: ["Class A", "Class B", "Class C"]
+      description: "Medical device software - software life cycle processes"
+    - name: "IEC 61508"
+      levels: ["SIL-1", "SIL-2", "SIL-3", "SIL-4"]
+      description: "Functional Safety of Electrical/Electronic/Programmable Electronic Safety-related Systems"
+
+# Tools organized alphabetically by name
+tools:
+
+  - name: "cargo"
+    type: "package-manager"
+    vendor: "-"
+    url: "https://github.com/rust-lang/cargo"
+    description: "Built-in Rust package manager and test runner"
+    license: "Apache 2.0 & MIT"
+    qualified: []
+
+  - name: "cargo-nextest"
+    type: "test-runner"
+    url: "https://nexte.st"
+    vendor: "-"
+    description: "Next-generation test runner for Rust"
+    license: "Apache 2.0 & MIT"
+    qualified: []
+
+  - name: "cargo-tarpaulin"
+    type: "code-coverage"
+    url: "https://github.com/xd009642/tarpaulin"
+    vendor: "-"
+    description: "Code coverage tool for Rust"
+    license: "Apache 2.0 & MIT"
+    qualified: []
+
+  - name: "clippy"
+    type: "static-analysis"
+    url: "https://github.com/rust-lang/rust-clippy"
+    vendor: "-"
+    description: "Clippy is a collection of lints to catch common mistakes and improve Rust code. It analyzes code for potential bugs, performance issues, style violations, and suggests more idiomatic Rust patterns."
+    license: "Apache 2.0 & MIT"
+    qualified: []
+
+  - name: "creusot"
+    type: "formal-verification"
+    url: "https://github.com/creusot-rs/creusot"
+    vendor: "-"
+    description: "Creusot is a deductive verifier for Rust code. It verifies your code is safe from panics, overflows, and assertion failures. By adding annotations you can take it further and verify your code does the correct thing."
+    license: "LGPL 2.1"
+    qualified: []
+
+  - name: "defmt-test"
+    type: "test-runner"
+    url: "https://github.com/knurling-rs/defmt"
+    vendor: "Ferrous Systems"
+    description: "Embedded testing framework"
+    license: "Apache 2.0 & MIT"
+    qualified: []
+
+  - name: "Ferrocene"
+    type: "qualified-compiler"
+    url: "https://ferrocene.dev"
+    vendor: "Ferrous Systems"
+    description: "Open-source qualified Rust compiler toolchain for safety- and mission-critical systems"
+    license: "commercial"
```

Ferrocene modifications to rustc retain the upstream license
```suggestion
    license: "Apache 2.0 & MIT"
```

---

`product` might be a bit confusing, because it somewhat overlaps with `source`.

maybe we name the field `qual-kit`, meaning qualification kit.
because commonly that is what customers must pay for even if the source is open source licensed and publicly available

---

thinking further on this, and inspired by your comment, I think we should leave license as plain text, and we can add an optional field, example

```yaml
# Ferrocene entry
license: "Apache 2.0 & MIT"
additional-details: binaries and support available with a monthly/yearly subscription, and qualification kit available for an additional fee
```

we can also leave "additional-details" off for now, to avoid slowing forward movement of this pr

---

as discussed in the last tooling meeting I added an optional `liability` field that should help to clarify such cases, but has a more important name compared to `license-note`